Poetry Install Issues with mise (Related to symlink and python3.*dylib missing)

I kept running into issues for installing Poetry using mise. 

### Local Setup To Reproduce
System: Mac OS (M1)

.tool-versions file:
```
poetry 1.6.1
```

Command to install and output:
```bash
mise install poetry 1.6.1 --verbose
[DEBUG] ARGS: /Users/<>/.local/bin/mise install poetry 1.6.1 --verbose
[DEBUG] Config {
    Config Files: [
        "~/.config/mise/config.toml",
        "~/.tool-versions",
    ],
}
[DEBUG] Toolset: go@1.21.5, python@3.11.6, rust@1.74.1
installing
~/.local/share/mise/plugins/poetry/bin/install
[DEBUG] $ ~/.local/share/mise/plugins/poetry/bin/install
Retrieving Poetry metadata
# Welcome to Poetry!
This will download and install the latest version of Poetry,
a dependency and package manager for Python.
It will add the `poetry` command to Poetry's bin directory, located at:
/Users/<>/.local/share/mise/installs/poetry/1.6.1/bin
You can uninstall at any time by executing this script with the --uninstall option,
and these changes will be reverted.
Installing Poetry (1.6.1)
Installing Poetry (1.6.1): Creating environment
Installing Poetry (1.6.1): An error occurred. Removing partial environment.
Poetry installation failed.
See /Users/<>/MI/poetry-installer-error-9usoojv5.log for error logs.
Cleanup: Something went wrong!
47 /Users/<>/.local/share/mise/plugins/poetry/bin/install: POETRY_HOME=$install_path python3 - --version "$version" $flags
[ERROR] ~/.local/share/mise/plugins/poetry/bin/install failed
Error:
   0: ~/.local/share/mise/plugins/poetry/bin/install exited with non-zero status: exit code 250

Location:
   /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/core/src/convert/mod.rs:757

Version:
   2024.2.2 macos-arm64 (1f11a1d 2024-02-02)

Suggestion: Run with --verbose or MISE_VERBOSE=1 for more information.

Backtrace omitted. Run with RUST_BACKTRACE=1 environment variable to display it.
Run with RUST_BACKTRACE=full to include source snippets.
```

Looking at the log file produced by poetry: 
```bash
cat /Users/<>/MI/poetry-installer-error-9usoojv5.log
dyld[32665]: Library not loaded: @executable_path/../lib/libpython3.11.dylib
  Referenced from: <EC57C3BD-D6B2-3EC2-8C9E-F9D225344939> /Users/ehunkler/.local/share/mise/installs/poetry/1.6.1/venv/bin/python3
  Reason: tried: '/Users/ehunkler/.local/share/mise/installs/poetry/1.6.1/venv/lib/libpython3.11.dylib' (no such file)

Traceback:

  File "<stdin>", line 923, in main
  File "<stdin>", line 562, in run
```

I googled a bunch and found this helpful thread: https://github.com/python-poetry/install.python-poetry.org/issues/24

The solution I found to fix this issue is I manually updated on my local machine the install script for poetry located here:

`~/.local/share/mise/plugins/poetry/bin/install`

and changed the install script to have the `sed 's/symlinks=False/symlinks=True/'` command. 

```
  if [ "$install_type" = "version" ]; then
    curl -sSL "$install_url" | sed 's/symlinks=False/symlinks=True/' | POETRY_HOME=$install_path python3 - --version "$version" $flags
  elif [ "$install_type" = "ref" ]; then
    curl -sSL "$install_url" | sed 's/symlinks=False/symlinks=True/' | POETRY_HOME=$install_path python3 - --git https://github.com/python-poetry/poetry.git@"$version" $flags
  else
    fail "unknown install type"
  fi
```

After these changes, the command worked for me and I was able to install poetry again with mise.

--- a/.github/workflows/workflow.yml
+++ b/.github/workflows/workflow.yml
@@ -20,15 +20,18 @@ jobs:
     steps:
       - name: Checkout
         uses: actions/checkout@v3
-      - uses: jdx/mise-action@v2
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        with:
-          cache: false
+      - run: |
+          curl mise.run | sh
+          echo 'export PATH=$HOME/.local/bin:$PATH' >> $GITHUB_ENV
       - run: mkdir -p ~/.local/share/mise/plugins
       - run: ln -s $PWD ~/.local/share/mise/plugins/poetry
-      - run: mise i poetry@latest
+      - run: mise i poetry@latest python
+        continue-on-error: true
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
       - run: mise x poetry@latest -- poetry --version
+      - run: cat poetry-installer-error-*.log && exit 1
+        if: failure()
 
   lint:
     runs-on: ubuntu-latest

--- a/bin/install
+++ b/bin/install
@@ -44,9 +44,9 @@ install_poetry() {
   fi
 
   if [ "$install_type" = "version" ]; then
-    curl -sSL "$install_url" | POETRY_HOME=$install_path python3 - --version "$version" $flags
+    curl -sSL "$install_url" | sed 's/symlinks=False/symlinks=True/' | POETRY_HOME=$install_path python3 - --version "$version" $flags
   elif [ "$install_type" = "ref" ]; then
-    curl -sSL "$install_url" | POETRY_HOME=$install_path python3 - --git https://github.com/python-poetry/poetry.git@"$version" $flags
+    curl -sSL "$install_url" | sed 's/symlinks=False/symlinks=True/' | POETRY_HOME=$install_path python3 - --git https://github.com/python-poetry/poetry.git@"$version" $flags
   else
     fail "unknown install type"
   fi

